velox/functions/prestosql/tests/IPAddressFunctionsTest.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+namespace facebook::velox::functions::prestosql {
+
+namespace {
+
+class IPAddressTest : public functions::test::FunctionBaseTest {
+ protected:
+  std::optional<std::string> castIPAddressVarcharCycle(
+      const std::optional<std::string> input) {
+    auto result = evaluateOnce<std::string>(
+        "cast(cast(c0 as ipaddress) as varchar)", input);
+    return result;
+  }
+
+  std::optional<std::string> castIPAddressVarbinary(
+      const std::optional<std::string> input) {
+    auto result = evaluateOnce<std::string>(
+        "cast(cast(from_hex(c0) as ipaddress) as varchar)", input);
```

Comment:
Is the extra cast to varchar required here ? cast(cast(from_hex(c0) as ipaddress) would return an int128_t. It might be worth testing for those values directly or have an external function for int128_t to a varchar representation of the IPAddress. I feel doing a cast to varchar in the CAST logic internally could hide some error cases.